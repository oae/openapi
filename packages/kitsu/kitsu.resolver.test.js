const { request } = require('graphql-request');
const { gql } = require('@openapi/core/utils');
const { createServer, destroyServer } = require('@openapi/core/testUtils');
const { name: pluginName } = require('./package.json');

let server = null;

beforeAll(async () => {
  server = await createServer([
    [
      pluginName,
      {
        auth: {
          clientId: process.env.OA_KITSU_CLIENT_ID,
          clientSecret: process.env.OA_KITSU_CLIENT_SECRET,
          username: process.env.OA_KITSU_USERNAME,
          password: process.env.OA_KITSU_PASSWORD,
        },
      },
    ],
  ]);
});

afterAll(async () => {
  await destroyServer(server);
});

const fragments = gql`
  fragment animeImageFields on AnimeImage {
    tiny
    small
    medium
    large
    original
  }

  fragment categoryFields on Category {
    title
    description
    slug
    nsfw
    cover {
      ...animeImageFields
    }
    createdAt
    updatedAt
  }

  fragment animeFields on Anime {
    title {
      en
      enJp
      jaJp
      canonicalTitle
    }
    slug
    synopsis
    createdAt
    updatedAt
    status
    averageRating
    startDate
    endDate
    posterImage {
      ...animeImageFields
    }
    coverImage {
      ...animeImageFields
    }
    episodeCount
    nsfw
    categories {
      ...categoryFields
    }
  }
`;

const validAnimes = {
  title: expect.toBeObject(),
  slug: expect.not.toBeEmpty(),
  synopsis: expect.not.toBeEmpty(),
  createdAt: expect.not.toBeEmpty(),
  updatedAt: expect.not.toBeEmpty(),
  status: expect.not.toBeEmpty(),
  averageRating: expect.toBeNumber(),
  startDate: expect.not.toBeEmpty(),
  endDate: expect.not.toBeEmpty(),
  posterImage: expect.toBeObject(),
  coverImage: expect.toBeObject(),
  episodeCount: expect.toBeNumber(),
  nsfw: expect.toBeBoolean(),
  categories: expect.toBeArray(),
};

describe('kitsu', () => {
  it('should return categories', async () => {
    const query = gql`
      {
        animeCategories(limit: 1) {
          title
          description
          slug
          nsfw
          cover {
            tiny
            small
            medium
            large
            original
          }
          createdAt
          updatedAt
        }
      }
    `;
    const result = await request(server.endpoint, query);
    expect(result).toMatchSnapshot();
  });

  it('should return animes', async () => {
    const query = gql`
      ${fragments}
      {
        animes(limit: 1) {
          ...animeFields
        }
      }
    `;
    const result = await request(server.endpoint, query);
    expect(result).toMatchSnapshot();
  });

  it('should return animes as far as limit', async () => {
    const query = gql`
      {
        animes(limit: 30) {
          nsfw
        }
      }
    `;
    const result = await request(server.endpoint, query);
    expect(result.animes).toHaveLength(30);
  });

  it('should return trending animes', async () => {
    const query = gql`
      ${fragments}
      {
        trendingAnimes {
          ...animeFields
        }
      }
    `;
    const result = await request(server.endpoint, query);
    expect(result.trendingAnimes).toBeArray();
    expect(result.trendingAnimes[0]).toMatchObject(validAnimes);
  });
});
