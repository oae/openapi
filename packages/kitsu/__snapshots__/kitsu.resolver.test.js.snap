// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kitsu should return animes 1`] = `
Object {
  "animes": Array [
    Object {
      "averageRating": 84.1,
      "categories": Array [
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:39:22.090Z",
          "description": "Science fiction stories usually focus on how situations could be different in the future when we are more technologically advanced; due to either our own achievements or having met with other civilizations whom we have learned from.",
          "nsfw": false,
          "slug": "science-fiction",
          "title": "Science Fiction",
          "updatedAt": "2017-05-31T09:01:35.589Z",
        },
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:38:47.070Z",
          "description": "The final frontier.",
          "nsfw": false,
          "slug": "space",
          "title": "Space",
          "updatedAt": "2017-05-31T09:05:33.687Z",
        },
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:39:28.357Z",
          "description": "Tragedy is a form of drama characterized by seriousness and dignity, usually involving a conflict between a character and some higher power, such as the law, the gods, fate, or society.",
          "nsfw": false,
          "slug": "tragedy",
          "title": "Tragedy",
          "updatedAt": "2017-05-31T09:01:35.721Z",
        },
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:39:20.357Z",
          "description": "Action anime usually involve a fairly straightforward story of good guys versus bad guys, where most disputes are resolved by using physical force. It often contains a lot of shooting, explosions and fighting.",
          "nsfw": false,
          "slug": "action",
          "title": "Action",
          "updatedAt": "2017-05-31T09:01:35.520Z",
        },
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:38:49.707Z",
          "description": "Space Travel usually refers going to space or travelling there with or without a spacecraft. There are different forms of space travel like Interplanetary travel, Interstellar travel or Intergalactic travel. Since it isn't possible to travel faster than light, most anime series involve some sort \\"sub-space travel mechanism\\" as a way to travel faster than light.",
          "nsfw": false,
          "slug": "space-travel",
          "title": "Space Travel",
          "updatedAt": "2017-05-31T06:39:37.583Z",
        },
        Object {
          "cover": Object {
            "large": "https://media.kitsu.io/categories/images/197/large.jpg?1496212770",
            "medium": "https://media.kitsu.io/categories/images/197/medium.jpg?1496212770",
            "original": "https://media.kitsu.io/categories/images/197/original.jpg?1496212770",
            "small": "https://media.kitsu.io/categories/images/197/small.jpg?1496212770",
            "tiny": "https://media.kitsu.io/categories/images/197/tiny.jpg?1496212770",
          },
          "createdAt": "2017-05-31T06:39:31.234Z",
          "description": "An anime that takes place on future Earth (or it could be an alternative universe) after a major cataclysm has occurred. World War III, an alien invasion, plague, global warming or some other general disaster are common themes to build a post-apocalyptic world upon, where the effects were severe enough that mankind barely survived.The time frame may be immediately after the catastrophe, focusing on the travails or psychology of survivors, or considerably later, often including the theme that the existence of pre-catastrophe civilization has been forgotten (or mythologised). Post-apocalyptic stories often take place in an agrarian, non-technological future world, or a world where only scattered elements of technology remain. There is a considerable degree of blurring between post-apocalyptic fiction and that which deals with false utopias or dystopian societies.",
          "nsfw": false,
          "slug": "post-apocalypse",
          "title": "Post Apocalypse",
          "updatedAt": "2018-07-18T16:06:09.683Z",
        },
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:38:50.800Z",
          "description": "The anime will fit this category if some or all of the action takes place on another planet which is supposedly in the same universe as Earth, not a complete fantasy world; usually there has to be some concept of space travel. The Latin name for Earth - Terra, would also pass the condition of being in the same universe as Earth. Basically, even mentioning of Earth existing or having existed would be enough. Although Moon is a moon, it also qualifies as Other Planet.",
          "nsfw": false,
          "slug": "other-planet",
          "title": "Other Planet",
          "updatedAt": "2017-05-31T06:39:37.833Z",
        },
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:38:47.047Z",
          "description": "Anime that fall under the future category are the ones that have a society that could have originated from ours. In most cases it should go hand to hand with sci-fi since we expect the future to consist of advanced technology, however, in the case where it's explicitly stated in the story that the anime takes place \\"x years into the future\\", the state of technology doesn't matter as long as it's long into the future. The anime can be set in an alternative universe not tied with our own, as long as it has a futuristic feel to it.

This category sports quite a lot of anime, mostly stemming from SciFi, but not necessarily only from there.",
          "nsfw": false,
          "slug": "future",
          "title": "Future",
          "updatedAt": "2017-05-31T09:05:33.654Z",
        },
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:38:50.836Z",
          "description": "Beam me up, Scotty! Space stations (think Death Star) also count. Basically, this category is to tell if the setting takes place onboard a spaceship in space. This category is not applicable if the spaceship lies on a planet and never actually gets into space.",
          "nsfw": false,
          "slug": "shipboard",
          "title": "Shipboard",
          "updatedAt": "2017-05-31T06:39:37.858Z",
        },
        Object {
          "cover": null,
          "createdAt": "2017-05-31T06:39:33.928Z",
          "description": null,
          "nsfw": false,
          "slug": "detective",
          "title": "Detective",
          "updatedAt": "2017-05-31T09:01:35.657Z",
        },
      ],
      "coverImage": Object {
        "large": "https://media.kitsu.io/anime/cover_images/1/large.jpg?1519178801",
        "medium": null,
        "original": "https://media.kitsu.io/anime/cover_images/1/original.jpg?1519178801",
        "small": "https://media.kitsu.io/anime/cover_images/1/small.jpg?1519178801",
        "tiny": "https://media.kitsu.io/anime/cover_images/1/tiny.jpg?1519178801",
      },
      "createdAt": "2013-02-20T16:00:13.609Z",
      "endDate": "1999-04-24T00:00:00.000Z",
      "episodeCount": 26,
      "nsfw": false,
      "posterImage": Object {
        "large": "https://media.kitsu.io/anime/poster_images/1/large.jpg?1431697256",
        "medium": "https://media.kitsu.io/anime/poster_images/1/medium.jpg?1431697256",
        "original": "https://media.kitsu.io/anime/poster_images/1/original.jpg?1431697256",
        "small": "https://media.kitsu.io/anime/poster_images/1/small.jpg?1431697256",
        "tiny": "https://media.kitsu.io/anime/poster_images/1/tiny.jpg?1431697256",
      },
      "slug": "cowboy-bebop",
      "startDate": "1998-04-03T00:00:00.000Z",
      "status": "finished",
      "synopsis": "In the year 2071, humanity has colonized several of the planets and moons of the solar system leaving the now uninhabitable surface of planet Earth behind. The Inter Solar System Police attempts to keep peace in the galaxy, aided in part by outlaw bounty hunters, referred to as \\"Cowboys\\". The ragtag team aboard the spaceship Bebop are two such individuals.
Mellow and carefree Spike Spiegel is balanced by his boisterous, pragmatic partner Jet Black as the pair makes a living chasing bounties and collecting rewards. Thrown off course by the addition of new members that they meet in their travels—Ein, a genetically engineered, highly intelligent Welsh Corgi; femme fatale Faye Valentine, an enigmatic trickster with memory loss; and the strange computer whiz kid Edward Wong—the crew embarks on thrilling adventures that unravel each member's dark and mysterious past little by little. 
Well-balanced with high density action and light-hearted comedy, Cowboy Bebop is a space Western classic and an homage to the smooth and improvised music it is named after.  
[Written by MAL Rewrite]",
      "title": Object {
        "canonicalTitle": "Cowboy Bebop",
        "en": "Cowboy Bebop",
        "enJp": "Cowboy Bebop",
        "jaJp": "カウボーイビバップ",
      },
      "updatedAt": "2018-11-13T20:23:10.471Z",
    },
  ],
}
`;

exports[`kitsu should return categories 1`] = `
Object {
  "animeCategories": Array [
    Object {
      "cover": null,
      "createdAt": "2017-05-31T06:38:28.636Z",
      "description": null,
      "nsfw": false,
      "slug": "middle-school",
      "title": "Middle School",
      "updatedAt": "2017-05-31T06:39:36.705Z",
    },
  ],
}
`;
